portability-worker/src/main/java/org/dataportabilityproject/worker/JobPollingService.java

```diff
@@ -119,7 +115,16 @@ private void keyGenerated(UUID jobId, KeyPair keyPair) throws IOException {
                     .setEncodedPublicKey(encodedPublicKey)
                     .setState(JobAuthorization.State.CREDS_ENCRYPTION_KEY_GENERATED)
                     .build()).build();
-    store.updateJob(jobId, updatedJob);
+    // Attempt to 'claim' this job by validating it is still in state CREDS_AVAILABLE as we
+    // update it to state CREDS_ENCRYPTION_KEY_GENERATED, along with our key. If another worker
+    // instance polled the same job, and already claimed it, it will have updated the job's state
+    // to CREDS_ENCRYPTION_KEY_GENERATED.
+    try {
+      store.updateJob(jobId, updatedJob, JobAuthorization.State.CREDS_AVAILABLE);
```

Comment:
I think we can ensure here that the existingJob state is CREDS_AVAILABLE rather than pass it into the JobStore